docs/source/development_guide/coding.rst

```diff
@@ -1,6 +1,27 @@
 Coding Conventions
 ==================
 
+Pre-Commit
+-----------
+
+.. _`Pre-Commit`: https://pre-commit.com/
+
+We use `pre-commit`_ to ensure that we're always keeping up with the best 
+practices when it comes to linting, standard code conventions and type 
+annotations. Although it may seem time consuming at first as to why is one 
+supposed to run all these tests and checks but it helps in identifying simple 
+issues before submission to code review. We've already specified a configuration 
+file with a list of hooks that will get executed before every commit. 
+
+First you'll need to setup the git hook scripts by installing them.
+
+.. code-block:: bash
+
```

Comment:
`pre-commit` is already present in `env.common.yml` so there shouldn't be any need to run `pip install pre-commit` right ?

Reply:
Hmm, I'm not sure actually. Could you try installing from source in a fresh environment following directions in the doc and see if it is present for you? The installation flow is a little complex so I'm not sure if the user will actually have pre-commit installed correctly on a fresh install

Reply:
Yes, it's present by default. Steps I followed:

- Clone my branch (`add-precommit`) to my local system and create a new conda env named `test` using `env.common.yml`.
- activate the `test` environment and run `pre-commit install`. PFA the output
```bash
pre-commit installed at .git/hooks/pre-commit
```
- run `pre-commit run --all-files` this runs pre-commit on the codebase. Output is as follows :
```bash
[INFO] Installing environment for https://github.com/pre-commit/mirrors-yapf.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
Check for merge conflicts................................................Passed
```

Reply:
PS: There seems to be a lot of `yapf` and `flake8` inconsistencies in the codebase.

Reply:
Ah good catch! If you're interested, a follow-up PR cleaning up yapf/flake8 inconsistencies in the codebase would be very useful